### PR TITLE
fix the thumnail size relative to viewport

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/7382

### Tests | QA Steps
1. Open any Chat. 
2. Send the attached image.
3. observe that image is limited to a fixed height when its original size is bigger than 40% of the screen height.

- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

image

#### Mobile Web

image

#### iOS

Updating in few hours....

#### Android

image

### CODE REVIEW

**Comment by parasharrajat** [comment]
Do we want to keep a minimum height for all images? For e.g. 40px takes from existing code. 

Otherwise, the image height could be very small based on the original upload image. 

try this image. 
image


Setting the minimum 40px height will show as. 
|Before|After|
|------------|-------|
|image|image|

**Comment by NikkiWines** [comment]
Yeah, I think a minimum height is good. Images that are long + skinny can be hard to click on because they're so small. Slack does something very similar to what you're suggesting as well 👍 

image

File: src/components/ThumbnailImage.js [hunk 1]
```diff
@@ -39,10 +42,20 @@ class ThumbnailImage extends PureComponent {
         // Width of the thumbnail works better as a constant than it does
         // a percentage of the screen width since it is relative to each screen
         // Note: Clamp minimum width 40px to support touch device
-        const thumbnailScreenWidth = lodashClamp(width, 40, 250);
+        let thumbnailScreenWidth = lodashClamp(width, 40, 250);
         const scaleFactor = width / thumbnailScreenWidth;
         const imageHeight = height / scaleFactor;
-        this.setState({thumbnailWidth: thumbnailScreenWidth, thumbnailHeight: imageHeight});
+        let thumbnailScreenHeight = lodashClamp(imageHeight, 40, this.props.windowHeight * 0.40);
+
```
Comment (NikkiWines) [inline src/components/ThumbnailImage.js:49]: Extra blank line

File: src/components/ThumbnailImage.js [hunk 2]
```diff
@@ -39,10 +42,20 @@ class ThumbnailImage extends PureComponent {
...
+        const aspectRatio = (height / width);
+
+        if (thumbnailScreenHeight > thumbnailScreenWidth) {
```
Comment (NikkiWines) [inline src/components/ThumbnailImage.js:52]: A comment here about what this is doing might be nice

File: src/components/ThumbnailImage.js [hunk 3]
```diff
@@ -39,10 +42,20 @@ class ThumbnailImage extends PureComponent {
         // Width of the thumbnail works better as a constant than it does
         // a percentage of the screen width since it is relative to each screen
         // Note: Clamp minimum width 40px to support touch device
-        const thumbnailScreenWidth = lodashClamp(width, 40, 250);
+        let thumbnailScreenWidth = lodashClamp(width, 40, 250);
         const scaleFactor = width / thumbnailScreenWidth;
```
Comment (NikkiWines) [inline src/components/ThumbnailImage.js:46]: NAB: scale factor doesn't really need to be its own variable. 

```Javascript
        const imageHeight = height / (width / thumbnailScreenWidth);

**Review by NikkiWines — CHANGES_REQUESTED** [review]
Some minor minor comments but overall looks good

**Review by NikkiWines — APPROVED** [review]
Looks good! One minor note, QA steps say "Send the attached image" but I don't see an attached image to send.

**Pull request merged**